checker/tests/guieffect/AnonInnerDefaults.java

```diff
@@ -57,13 +106,108 @@ public void doStuff() {
+                };
+        @UI PolyIface p4 =
+                new IndirectPolyIface() {
+                    @Override
+                    public void doStuff() {
+                        // Safe due to anonymous inner class effect inference
+                        e.dangerous(); // should be okay
+                    }
+                };
+        @UI ParlyPolyIface p5 =
+                new ParlyPolyIface() {
+                    @Override
+                    public void doPolyUIStuff() {
+                        // Safe due to anonymous inner class effect inference
+                        e.dangerous(); // should be okay
+                    }
+
+                    @Override
+                    @SafeEffect
+                    public void doSafeStuff() {
+                        e.repaint();
+                    }
+                };
+        ParlyPolyIface p6 =
+                new ParlyPolyIface() {
+                    @Override
+                    public void doPolyUIStuff() {
+                        e.repaint(); // Safe
+                    }
+
+                    @Override
+                    public void doSafeStuff() {
                         // :: error: (call.invalid.ui)
+                        e.dangerous(); // No inference here, just as an invalid call
+                    }
+                };
+        @UI ParlyPolyIface p7 =
+                new ParlyPolyIface() {
+                    @Override
+                    public void doPolyUIStuff() {
+                        // Safe due to anonymous inner class effect inference
                         e.dangerous(); // should be okay
                     }
+
+                    @Override
+                    @SafeEffect
+                    public void doSafeStuff() {
+                        // :: error: (call.invalid.ui)
+                        e.dangerous(); // No inference here, just as an invalid call
+                    }
                 };
+        callSafePolyIface(
+                // :: error: (argument.type.incompatible)
+                new PolyIface() { // Anonymous inner class inference for @UI
+                    @Override
+                    public void doStuff() {
+                        // Safe due to anonymous inner class effect inference
+                        e.dangerous(); // should be okay
+                    }
+                });
+        callUIPolyIface(
+                new PolyIface() { // Anonymous inner class inference for @UI
+                    @Override
+                    public void doStuff() {
+                        // Safe due to anonymous inner class effect inference
+                        e.dangerous(); // should be okay
+                    }
+                });
+        callSafePolyIface(getSafePolyIface(e));
+        callUIPolyIface(getUIPolyIface(e));
+
+        // ToDo: To decide, this case and whether or not we base on @PolyUIType for super type or @PolyUIEffect for
+        // overriden method.
```

Comment:
Was this decided, or is this TODO stale?

Reply:
No clue why that TODO is here, of all places. It has nothing to do with the next test case. Removing it now. But, for the record, this was decided to be based on overriden method according to whatever `AnnotatedTypes.overriddenMethods` returns 😃 (but the declaring class must also be annotated `@PolyUIType`, we are fairly conservative about turning on inference)